### ISSUE TITLE
✨ Enhancement: Api keys and other secrets should be loaded in env instead of hardcoding them in source

### ISSUE THREAD

**Issue by maverox** [json-schema-org/website#514]
### Is your feature request related to a problem? Please describe

Not necessarily but i think it can result into a problem as it's not a good practice i presume for apiKeys or any keys to remain out int the open.

### Describe the solution you'd like

to use dotenv package to load them from an env file

### Describe alternatives you've considered

using env files to store such variables 
and use dotenv to load them

### Are you working on this?

Yes

**Comment by maverox** [comment]
Arey you suggesting i change my issue and make it less specific?
because i didn't exposed the secret itself
But i will do as you say

**Comment by gregsdennis** [comment]
> Arey you suggesting i change my issue and make it less specific?
> because i didn't exposed the secret itself

No, but you exposed where to find it.

**Comment by maverox** [comment]
I have removed those specifics from my message, and also informed him.
I was just trying to help. 

Can you please guide me as to what i should do now?

**Comment by benjagm** [comment]
Thanks @maverox ! As you said it will be great to avoid this practice. In order to fix this we need to find a way of:

- Use .env file.
- Remove any hardcoded api key.
- Make sure we can upload the production .env during our deployments in cloudflare.

**Comment by maverox** [comment]
I have attached pr #515 that hopefully covers your first two agenda points.
Please have a look and guide me!

**Comment by maverox** [comment]
Also Can you please make assign this to me and make this the mandatory task,
Mentioned in contributors guide for gsoc?

**Comment by maverox** [comment]
> > I have attached pr [#515](https://github.com/json-schema-org/website/pull/515) that hopefully covers your first two agenda points.
> > Please have a look and guide me!
> 
> Your PR looks great but we need to fix and implement the 3 steps at the same time. Open to do some research for point 3?

Yes I will gladly do, i hope there's no hard deadline as it might take a while.

**Comment by maverox** [comment]
@benjagm  any advise on further steps that i might take, i am stuck at this moment.

**Comment by benjagm** [comment]
I know you are stuck Uzair. But unfortunately we can proceed with merging this until we have the cloudflare worked. 

You have done great. This count as qualification task in this current state so dont worry. As soon as we have time for work the cloudflare part will complete this.

**Comment by aialok** [comment]
Please assign this issue to me as I am currently working on it. 
Thank you!

**Comment by benjagm** [comment]
Thanks Alok. All yours.

@maverox Huge thanks for the work done with this. Now finally are actively working on this again.

**Comment by gregsdennis** [comment]
I thought I had mentioned this before, but it doesn't appear so:  Creating and setting secrets can only be done by an admin.  Updating the GH action YAML can be done by anyone, but this particular task will need to be performed by one of us.

**Comment by aialok** [comment]
Yes, most of the work has been done already. We don't really need to update anything in YAML or GitHub Action secrets. Setting up .env can be done by the admin who has access to the Cloudflare website project. 
From my side, I need to update documentation of installation regarding the uses of .env while setting up project.


![Screenshot from 2024-05-14 13-10-28](https://github.com/json-schema-org/website/assets/66772290/f27c3663-3911-4581-9990-5c62ac494119)

**Comment by benjagm** [comment]
I just created the variables in cloudflare:

<img width="713" alt="Screenshot 2024-05-14 at 13 53 09" src="https://github.com/json-schema-org/website/assets/40007659/59b57a87-28d4-49fd-8f41-c04b1cdb1ed2">

**Comment by aialok** [comment]
Great ! I checkout the PR in my local everything working fine. We also need to dump the exposed api key so that others will not be able exploit it.  

Thank you  : )

**Comment by maverox** [comment]
Have I been unassigned because of inactivity?
Nonetheless, thank you @aialok  for taking it up and continuing with the work!🙌.

**Comment by benjagm** [comment]
@maverox My apologies you are not unassigned!! As part of GSoC Alok is working in the CI CD project and finally we can integrate the great work you did. 

You are completing the final step together!

**Comment by maverox** [comment]
That is really exciting, happy to be of any use.
Best of luck @aialok.

**Comment by aialok** [comment]
Thank you @maverox for amazing work you have done : )
